content/blog/2019-02-18-stats19.md

```diff
@@ -0,0 +1,1014 @@
+Lets read the datasets first:
+
+``` r
+#crashes_2017 = get_stats19(year = 2017, type = "Accidents", ask = FALSE)
+casualties_2017 = get_stats19(year = 2017, type = "Casualties", ask = FALSE)
+nrow(casualties_2017)
+```
+
+    ## [1] 170993
+
+``` r
+vehicles_2017 = get_stats19(year = 2017, type = "Vehicles", ask = FALSE)
+nrow(vehicles_2017)
+```
+
+    ## [1] 238926
+
+Lets now read in casualties that took place in West Yorkshire (using
+`crashes_wy` object above), and count the number of casualties by
+severity for each crash:
+
+``` r
+library(tidyr)
+library(dplyr)
+sel = casualties_2017$accident_index %in% crashes_wy$accident_index
+casualties_wy = casualties_2017[sel, ]
+cas_types = casualties_wy %>% 
+  select(accident_index, casualty_type) %>% 
+  group_by(accident_index) %>% 
+  summarise(
+    Total = n(),
+    walking = sum(casualty_type == "Pedestrian"),
+    cycling = sum(casualty_type == "Cyclist"),
+    passenger = sum(casualty_type == "Car occupant")
+    ) 
+cj = left_join(crashes_wy, cas_types)
+```
+
+What just happened?
+
+We found the subset of casualties that took place in West Yorkshire with
+reference to the `accident_index` variable in the `accidents` table.
+Then we used the **dplyr** function `summarise()`, to find the number of
+people who were in a car, cycling, and walking when they were injured.
+This new casualty dataset is joined onto the `crashes_wy` dataset. The
+result is a spatial (`sf`) data frame of crashes in West Yorkshire, with
+columns counting how many road users of different types were hurt. The
+joined data has additional variables:
+
+``` r
+base::setdiff(names(cj), names(crashes_wy))
+```
+
+    ## [1] "Total"     "walking"   "cycling"   "passenger"
+
+In addition to the `Total` number of people hurt/killed, `cj` contains a
+column for each type of casualty (cyclist, car occupant, etc.), and a
+number corresponding to the number of each type hurt in each crash. It
+also contains the `geometry` column from `crashes_sf`. In other words,
+joins allow the casualties and vehicles tables to be geo-referenced. We
+can then explore the spatial distribution of different casualty types.
+The following figure, for example, shows the spatial distribution of
+pedestrians and car passengers hurt in car crashes across West Yorkshire
+in 2017:
+
+``` r
+library(ggplot2)
+crashes_types = cj %>% 
+  filter(accident_severity != "Slight") %>% 
+  mutate(type = case_when(
+    walking > 0 ~ "Walking",
+    cycling > 0 ~ "Cycling",
+    passenger > 0 ~ "Passenger",
+    TRUE ~ "Other"
+  ))
+ggplot(crashes_types, aes(size = Total, colour = speed_limit)) +
+  geom_sf(show.legend = "point", alpha = 0.3) +
+  facet_grid(vars(type), vars(accident_severity)) +
+  scale_size(
+    breaks = c(1:3, 12),
+    labels = c(1:2, "3+", 12)
+    ) +
+  scale_color_gradientn(colours = c("blue", "yellow", "red")) +
+  theme(axis.text = element_blank(), axis.ticks = element_blank())
+```
+
```

Comment:
It would be helpful to add a sentence before or after every figure to state the conclusion a reader would make. Consider adding the `title=` text to the text of the post

And for the Great Britain map above add e.g. shows that the largest number of fatalities per police force is in London (?)

Reply:
I did, not sure why it was not there. Thanks